### ISSUE TITLE
[Feature Request]: Possible to work with new Google Doorbell (battery/wired)

### ISSUE THREAD

**Issue by djjuice** [Brandawg93/homebridge-nest-cam#440]
### Enhancement of issue

Knowing that the new doorbell only works in the google home app, the plugin obviously doesnt see the camera since it's not added to the nest app. the new nest indoor/outdoor cameras appear to be moving this way as well.

### Solution

If possible would like to see the camera from the google home app,

### Additional Context

I totally understand if this is outside the scope of this plug-in but at least wanted to ask.

**Comment by Brandawg93** [comment]
Correct. The new Nest cameras (including the new doorbell) do *not* work with this plugin. They do not connect to the Nest app which is what this plugin uses to interface with the cameras.

I wouldn't say it is entirely out-of-scope of this plugin but would definitely be a major rewrite. I assume these new cameras will be able to interface with [Matter](https://buildwithmatter.com/) soon, but from what I understand, older cameras will not be capable of using Matter, and this plugin will always be around to circumvent that.

**Comment by Nowie86** [comment]
Will there be a new plugin for the new doorbell and battery-powered cameras?
Thanks

**Comment by Brandawg93** [comment]
I am not working on it at the moment. Matter may make a new plugin pointless.

**Comment by limitless00net** [comment]
I've found new API for Google Nest Doorbell (battery).
https://developers.google.com/nest/device-access/api/doorbell-battery

Also, I know camera device type is not defined by Matter.

**Comment by Nowie86** [comment]
Also the nest cam battery version

https://developers.google.com/nest/device-access/api/camera-battery

**Comment by kilgil27** [comment]
The starling hub is able to integrate the new cameras, so hopefully this is possible with homebridge as well

**Comment by Brandawg93** [comment]
I looked into this a bit. 

There are two major refactors that would need to take place. 
1. The transition to SDM. The only problem is that SDM is not feature complete so some things like talkback will not work. 
2. The addition of a WebRTC controller. All new cameras use WebRTC instead of RTSP to relay the camera stream. 

This would be quite a major rewrite. I'm sure the Starling team has spent many hours to get this working and I commend them! Unfortunately, It's a little more than I want to commit to currently. 

Tl;dr - It _can_ be done but it will take a _lot_ of work.

**Comment by brucejohnson0** [comment]
Are you sure about SDM not supporting talkback? The starling hub definitely supports talkback with the new cameras. Maybe they don't use SDM but use some other reverse engineered API.

**Comment by Brandawg93** [comment]
If Starling has talkback then it doesn't use SDM. Maybe it's using the gRPC protocols that the Google Home app uses? That's interesting to know. Thanks for sharing!

**Comment by jricey** [comment]
> If Starling has talkback then it doesn't use SDM. Maybe it's using the gRPC protocols that the Google Home app uses? That's interesting to know. Thanks for sharing!

Hey Hey, have you had any luck with integrating this with the Google Nest Doorbell?

**Comment by nhatvcao1994** [comment]
With iOS 16 announcement and Matter support, this 2021 Battery Doorbell should be supported pretty soon I hope?

**Comment by jfroy** [comment]
> With iOS 16 announcement and Matter support, this 2021 Battery Doorbell should be supported pretty soon I hope?

I've seen no reports that Google will provide a firmware update to add Matter support to that product. Additionally, the initial version of Matter is not expected to support cameras[^1].

[^1]: https://www.consumerreports.org/smart-home/matter-smart-home-standard-faq-a9475777045/
